wrappers/reason.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2014-2019 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the license is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package wrappers
+
+const (
+	SHTDN_REASON_FLAG_USER_DEFINED = 0x40000000
+	SHTDN_REASON_FLAG_PLANNED      = 0x80000000
+)
+
+const (
+	SHTDN_REASON_MAJOR_OTHER           = 0x00000000
+	SHTDN_REASON_MAJOR_NONE            = 0x00000000
+	SHTDN_REASON_MAJOR_HARDWARE        = 0x00010000
+	SHTDN_REASON_MAJOR_OPERATINGSYSTEM = 0x00020000
+	SHTDN_REASON_MAJOR_SOFTWARE        = 0x00030000
+	SHTDN_REASON_MAJOR_APPLICATION     = 0x00040000
+	SHTDN_REASON_MAJOR_SYSTEM          = 0x00050000
+	SHTDN_REASON_MAJOR_POWER           = 0x00060000
+	SHTDN_REASON_MAJOR_LEGACY_API      = 0x00070000
+)
+
+const (
+	SHTDN_REASON_MINOR_OTHER                 = 0x00000000
+	SHTDN_REASON_MINOR_NONE                  = 0x000000ff
+	SHTDN_REASON_MINOR_MAINTENANCE           = 0x00000001
+	SHTDN_REASON_MINOR_INSTALLATION          = 0x00000002
+	SHTDN_REASON_MINOR_UPGRADE               = 0x00000003
+	SHTDN_REASON_MINOR_RECONFIG              = 0x00000004
+	SHTDN_REASON_MINOR_HUNG                  = 0x00000005
+	SHTDN_REASON_MINOR_UNSTABLE              = 0x00000006
+	SHTDN_REASON_MINOR_DISK                  = 0x00000007
+	SHTDN_REASON_MINOR_PROCESSOR             = 0x00000008
+	SHTDN_REASON_MINOR_NETWORKCARD           = 0x00000009
+	SHTDN_REASON_MINOR_POWER_SUPPLY          = 0x0000000a
+	SHTDN_REASON_MINOR_CORDUNPLUGGED         = 0x0000000b
+	SHTDN_REASON_MINOR_ENVIRONMENT           = 0x0000000c
+	SHTDN_REASON_MINOR_HARDWARE_DRIVER       = 0x0000000d
+	SHTDN_REASON_MINOR_OTHERDRIVER           = 0x0000000e
+	SHTDN_REASON_MINOR_BLUESCREEN            = 0x0000000F
+	SHTDN_REASON_MINOR_SERVICEPACK           = 0x00000010
+	SHTDN_REASON_MINOR_HOTFIX                = 0x00000011
+	SHTDN_REASON_MINOR_SECURITYFIX           = 0x00000012
+	SHTDN_REASON_MINOR_SECURITY              = 0x00000013
+	SHTDN_REASON_MINOR_NETWORK_CONNECTIVITY  = 0x00000014
+	SHTDN_REASON_MINOR_WMI                   = 0x00000015
+	SHTDN_REASON_MINOR_SERVICEPACK_UNINSTALL = 0x00000016
+	SHTDN_REASON_MINOR_HOTFIX_UNINSTALL      = 0x00000017
+	SHTDN_REASON_MINOR_SECURITYFIX_UNINSTALL = 0x00000018
+	SHTDN_REASON_MINOR_MMC                   = 0x00000019
+	SHTDN_REASON_MINOR_SYSTEMRESTORE         = 0x0000001a
+	SHTDN_REASON_MINOR_TERMSRV               = 0x00000020
+	SHTDN_REASON_MINOR_DC_PROMOTION          = 0x00000021
+	SHTDN_REASON_MINOR_DC_DEMOTION           = 0x00000022

```

Comment:
It appears that `SHTDN_REASON_MAJOR_NONE`, `SHTDN_REASON_MINOR_NONE`, `SHTDN_REASON_MINOR_SYSTEMRESTORE`, `SHTDN_REASON_MINOR_DC_PROMOTION`, and `SHTDN_REASON_MINOR_DC_DEMOTION` are undocumented, so let's remove them from here.  See https://docs.microsoft.com/en-us/windows/desktop/shutdown/system-shutdown-reason-codes.